examples/advanced/float_32_64_benchmark.py

```diff
@@ -0,0 +1,168 @@
+        ENa = 50 * mV
+        g_na = (100 * msiemens * cm ** -2) * area
+        g_kd = (30 * msiemens * cm ** -2) * area
+        VT = -63 * mV
+        # Time constants
+        taue = 5 * ms
+        taui = 10 * ms
+        # Reversal potentials
+        Ee = 0 * mV
+        Ei = -80 * mV
+        we = 6 * nS  # excitatory synaptic weight
+        wi = 67 * nS  # inhibitory synaptic weight
+
+        # The model
+        eqs = Equations('''
+        dv/dt = (gl*(El-v)+ge*(Ee-v)+gi*(Ei-v)-
+                 g_na*(m*m*m)*h*(v-ENa)-
+                 g_kd*(n*n*n*n)*(v-EK))/Cm : volt
+        dm/dt = alpha_m*(1-m)-beta_m*m : 1
+        dn/dt = alpha_n*(1-n)-beta_n*n : 1
+        dh/dt = alpha_h*(1-h)-beta_h*h : 1
+        dge/dt = -ge*(1./taue) : siemens
+        dgi/dt = -gi*(1./taui) : siemens
+        alpha_m = 0.32*(mV**-1)*(13*mV-v+VT)/
+                 (exp((13*mV-v+VT)/(4*mV))-1.)/ms : Hz
+        beta_m = 0.28*(mV**-1)*(v-VT-40*mV)/
+                (exp((v-VT-40*mV)/(5*mV))-1)/ms : Hz
+        alpha_h = 0.128*exp((17*mV-v+VT)/(18*mV))/ms : Hz
+        beta_h = 4./(1+exp((40*mV-v+VT)/(5*mV)))/ms : Hz
+        alpha_n = 0.032*(mV**-1)*(15*mV-v+VT)/
+                 (exp((15*mV-v+VT)/(5*mV))-1.)/ms : Hz
+        beta_n = .5*exp((10*mV-v+VT)/(40*mV))/ms : Hz
+        ''')
+
+        P = NeuronGroup(4000, model=eqs, threshold='v>-20*mV', refractory=3 * ms,
+                        method='exponential_euler')
+        Pe = P[:3200]
+        Pi = P[3200:]
+        Ce = Synapses(Pe, P, on_pre='ge+=we')
+        Ci = Synapses(Pi, P, on_pre='gi+=wi')
+        Ce.connect(p=0.02)
+        Ci.connect(p=0.02)
+
+        # Initialization
+        P.v = 'El + (randn() * 5 - 5)*mV'
+        P.ge = '(randn() * 1.5 + 4) * 10.*nS'
+        P.gi = '(randn() * 12 + 20) * 10.*nS'
+
+    run(1 * second, profile=True)
+
+    return sum(t for name, t in magic_network.profiling_info)
+
+def generate_results(num_repeats):
+    results = {}
+
+    for name in ['CUBA', 'COBA']:
+        for target in ['numpy', 'cython', 'weave']:
+            for dtype in [float32, float64]:
+                prefs.codegen.target = target
+                prefs.core.default_float_dtype = dtype
+                times = [run_benchmark(name) for repeat in range(num_repeats)]
+                results[name, target, dtype.__name__] = amin(times)
+
+    for name in ['CUBA', 'COBA']:
+        for dtype in [float32, float64]:
+            times = []
+            for _ in range(num_repeats):
+                reset_device()
+                reinit_devices()
+                set_device('cpp_standalone', directory=None, with_output=False)
+                prefs.core.default_float_dtype = dtype
+                times.append(run_benchmark(name))
+            results[name, 'cpp_standalone', dtype.__name__] = amin(times)
+
+    return results
+
+# Cache this if we have joblib installed, but don't worry if not
+try:
+    import joblib
+    mem = joblib.Memory(location='.', verbose=0)
```

Comment:
In older joblib versions, the first argument is called `cachedir` not `location`, this can therefore fail with a `TypeError`. Simply leaving away the argument name should work for all versions.

Reply:
Maybe I should just remove the joblib stuff actually? It's nice to have for developing the script but not that useful in the end.

Reply:
Good point, I agree that it's probably better to keep this as simple as possible.

Reply:
Done!

Reply:
Got distracted while filling out the commit message so it's a bit cryptic